projects/dashboards/src/widgets/conditional/conditional.model.ts

```diff
@@ -0,0 +1,34 @@
+import { Model, ModelApi, ModelJson, ModelProperty, PLAIN_OBJECT_PROPERTY } from '@hypertrace/hyperdash';
+import { ModelInject, MODEL_API } from '@hypertrace/hyperdash-angular';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+@Model({
+  type: 'conditional'
+})
+export class ConditionalModel {
+  @ModelProperty({
+    key: 'true',
+    type: PLAIN_OBJECT_PROPERTY.type,
+    required: true
+  })
+  public true!: ModelJson;
+
+  @ModelProperty({
+    key: 'false',
+    type: PLAIN_OBJECT_PROPERTY.type,
+    required: true
+  })
+  public false!: ModelJson;
+
+  @ModelInject(MODEL_API)
+  private readonly api!: ModelApi;
+
+  public getData(): Observable<boolean> {
+    return this.api.getData<boolean>();
+  }
+
+  public getModel(): Observable<ModelJson> {
+    return this.getData().pipe(map(result => (result ? this.true : this.false)));
```

Comment:
Based on how we're using this in the renderer, we should hydrate this modelJson, and return a real model - something like `this.api.createChild(result ? this.true : this.false)`